src/drivers/imx/sdma.c

```diff
@@ -579,23 +580,31 @@ static int sdma_release(struct dma_chan_data *channel)
 
 static int sdma_copy(struct dma_chan_data *channel, int bytes, uint32_t flags)
 {
-	/* Check the buffer descriptors, set the "DONE" bit on one of
-	 * them. Ignoring "bytes" value, if the pipeline is incorrect it
-	 * will be audible.
-	 */
 	struct sdma_chan *pdata = dma_chan_get_data(channel);
+	struct dma_cb_data next = {
+		.channel = channel,
+		.elem.size = bytes,
+	};
 
 	tracev_sdma("sdma_copy");
+
+	pdata->next_bd = (pdata->next_bd + 1) % 2;
+
 	for (int i = 0; i < pdata->desc_count; i++) {
 		/* Work around the fact that we cannot allocate uncached memory
 		 * on all platforms supporting SDMA.
 		 */
 		dcache_invalidate_region(&pdata->desc[i].config,
 					 sizeof(pdata->desc[i].config));
-		pdata->desc[i].config |= SDMA_BD_DONE;
+		if (i == pdata->next_bd)
+			pdata->desc[i].config |= SDMA_BD_DONE;
+
```

Comment:
Can't we get rid of the loop overall?

Reply:
Yes, this can be done use directly pdata->next_bd as the index of desc to be modified.

Reply:
@paulstelian97 OK, fixed!